buffer.go

```diff
@@ -318,6 +319,11 @@ func (b *buffer) resize(size int) {
 		if bufferSize < minSize {
 			bufferSize = minSize
 		}
+		// put old byte slice back in the pool
+		buffer := newBuffer(b.data)
+		buffer.pool = b.pool
+		buffer.unref() // unref will put it back in the pool
+
```

Comment:
Did you notice that this was significantly improving memory reuse? I'm asking to make sure it's worth the complexity (e.g. instead of leaving the GC take care of reclaiming memory), as there are other cases where we append to the buffers byte slices for example which result in GC work anyways.

Reply:
I did not test this change independently. Since we're planning on just removing `resize()` then this should go away.